### PR TITLE
Circumcision

### PR DESCRIPTION
**Pull request by robynstuart**
@cliffckerr please review

### CODE REVIEW

File: optima/model.py [hunk 1]
```diff
@@ -379,10 +374,16 @@ def model(simpars=None, settings=None, verbose=None, benchmark=False, die=True):
             pop2 = this['pop2']
             thistrans = this['trans']
             
-            thisforceinf = 1 - mpow((1-thistrans*circeff[pop1,t]*prepeff[pop1,t]*stieff[pop1,t]), (dt*cond*acts*effhivprev[pop2]))
-            forceinfvec[pop1] = 1 - (1-forceinfvec[pop1]) * (1-thisforceinf)  
-            
-            if not(forceinfvec[pop1]>=0):
+            if male[pop1]: # Separate FOI calcs for circs vs uncircs
+                thisforceinf_uncirc = 1 - mpow((1-thistrans*prepeff[pop1,t]*stieff[pop1,t]), (dt*cond*acts*effhivprev[pop2]))
+                thisforceinf_circ = 1 - mpow((1-thistrans*circeff[pop1,t]*prepeff[pop1,t]*stieff[pop1,t]), (dt*cond*acts*effhivprev[pop2]))
```
Comment (cliffckerr) [inline optima/model.py]: this could be done with a loop but it's a bit complicated to get the conditionals right so will leave for now

Comment (robynstuart) [inline optima/model.py]: agreed

File: optima/parameters.py [hunk 2]
```diff
@@ -373,8 +372,8 @@ def makepars(data, label=None, verbose=2):
             pars['initprev'] = data2prev(data=data, keys=keys, **rawpar) # Pull out first available HIV prevalence point
         elif partype=='popsize': # Population size only
             pars['popsize'] = data2popsize(data=data, keys=keys, **rawpar)
-        elif partype=='timepar': # It's a normal time parameter, e.g. hivtest
-            pars[parname] = data2timepar(data=data, keys=keys, **rawpar)
+        elif partype=='timepar':            
+            pars[parname] = data2timepar(data=data, keys=keys, **rawpar) # Otherwise it's a regular time par, made from data
```
Comment (cliffckerr) [inline optima/parameters.py]: i don't quite understand the rationale for this...?

Comment (robynstuart) [inline optima/parameters.py]: ha. you change a thing, you change it back, you change it back again, you forget how it was to begin with, you can't go on, you won't go on, you'll go on.

File: optima/scenarios.py [hunk 3]
```diff
@@ -130,7 +130,8 @@ def makescenarios(project=None, scenlist=None, verbose=2):
                     else: 
                         errormsg = 'Unrecognized population or partnership type.'
                         raise OptimaException(errormsg)
-                    for pop in pops:
+
+                    for popno, pop in enumerate(pops):
```
Comment (cliffckerr) [inline optima/scenarios.py]: personally i prefer length of variable name to roughly correspond to length of variable -- e.g. 'popno' is e.g. 1 while 'pop' could be 'M 50+', so i'd usually go for something like `for p,pop`, but obviously not a big deal, and i violate myself all the time with
`nthings = len(things)`

**Pull request merged**